File: src/audio/playlist/sequence/Selection.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package audio.playlist.sequence;

import java.util.List;

/**
 * Selection logic for selection of next and previous item in a list. Used in
 * combination with {@link ItemSelector}.
 * <p>
 * @author Plutonium_
 */
@FunctionalInterface
public interface Selection<E> {

    /**
     * Selects an item from list.
     *
     * @param size  Size of the list.
     * @param index Index of currently selected item.
     * @param item Currently selected item.
     * @param list List the selection operates on.
     * @return
     */
    public E select(int size, int index, E item, List<E> list);

}

File: src/player/main/sp/it/pl/audio/playlist/sequence/Selection.kt
package sp.it.pl.audio.playlist.sequence

/** Selection logic for selection of next and previous item in a list. */
interface Selection<E> {

   /**
    * Selects an item from list.
    *
    * @param size size of the list.
    * @param index index of the currently selected item.
    * @param item currently selected item.
    * @param list list the selection operates on.
    */
   fun select(size: Int, index: Int, item: E, list: List<E>): E

}